### PR TITLE
protect against panic if taint does not have timestamp

### PR DESCRIPTION
**Pull request by dbenque**
The field `timeAdded` in the taint is a pointer and if a user or a program does not set it correctly our code could panic in couple of places.
This PR added protections so that we report error or have default behavior instead of panic.

Note: in theory the taints that we are using are only set "correctly" by us... but it is better to protect ourselves against possible problem introduced by external users/processes

### CODE REVIEW

File: internal/drain_runner/pre_processor/pre_activities.go [hunk 1]
```diff
@@ -95,6 +95,10 @@ func (pre *PreActivitiesPreProcessor) IsDone(ctx context.Context, node *corev1.N
 			pre.eventRecorder.NodeEventf(ctx, node, corev1.EventTypeWarning, eventPreActivityFailed, "pre activity '%s' failed", key)
 			return false, PreProcessNotDoneReasonFailure, nil
 		default:
+			if taint.TimeAdded == nil {
+				return false, PreProcessNotDoneReasonProcessing, fmt.Errorf("found 'drained' taint without timeAdded field set")
```
Comment (dhenkel92) [inline internal/drain_runner/pre_processor/pre_activities.go:99]:
```suggestion
				return false, PreProcessNotDoneReasonProcessing, fmt.Errorf("found 'drain-candidate' taint without timeAdded field set")
```

File: internal/drain_runner/pre_processor/pre_processor.go [hunk 2]
```diff
@@ -55,6 +56,10 @@ func (pre *WaitTimePreprocessor) IsDone(ctx context.Context, node *corev1.Node)
 		return true, "", nil
 	}
 
+	if taint.TimeAdded == nil {
+		return false, PreProcessNotDoneReasonProcessing, fmt.Errorf("found 'drained' taint without timeAdded field set")
```
Comment (dhenkel92) [inline internal/drain_runner/pre_processor/pre_processor.go:60]:
```suggestion
		return false, PreProcessNotDoneReasonProcessing, fmt.Errorf("found 'drain-candidate' taint without timeAdded field set")
```

File: internal/drain_runner/runner.go [hunk 3]
```diff
@@ -146,6 +146,10 @@ func (runner *drainRunner) handlePendingDrainedNodes(ctx context.Context, info *
 		if !exist || taint.Value != k8sclient.TaintDrained {
 			continue
 		}
+		if taint.TimeAdded == nil {
+			runner.logger.Error(fmt.Errorf("found 'drained' taint without timeAdded field set"), "node", n.Name)
```
Comment (dhenkel92) [inline internal/drain_runner/runner.go:150]:
```suggestion
			runner.logger.Error(fmt.Errorf("found 'drain-candidate' taint without timeAdded field set"), "node", n.Name)
```

**Review by dhenkel92 — APPROVED** [review]

File: internal/drain_runner/pre_processor/pre_activities.go (hunk 1 above)
Comment (dbenque) [reply internal/drain_runner/pre_processor/pre_activities.go]: done

File: internal/drain_runner/pre_processor/pre_processor.go (hunk 2 above)
Comment (dbenque) [reply internal/drain_runner/pre_processor/pre_processor.go]: done

File: internal/drain_runner/runner.go (hunk 3 above)
Comment (dbenque) [reply internal/drain_runner/runner.go]: I took the 2 others, but I think this one is correct.
That was my source for the unfortunate copy/paste in fact :)

Comment (dhenkel92) [reply internal/drain_runner/runner.go]: ah, right

**Review by dhenkel92 — APPROVED** [review]

**Pull request merged**